x/incentives/client/cli/query.go

```diff
@@ -361,45 +363,88 @@ func GetCmdRewardsEst() *cobra.Command {
+			lockIdsCombined, err := cmd.Flags().GetString(FlagLockIds)
 			if err != nil {
 				return err
 			}
+			lockIdStrs := strings.Split(lockIdsCombined, ",")
 
-			lockIdsCombined, err := cmd.Flags().GetString(FlagLockIds)
+			endEpoch, err := cmd.Flags().GetInt64(FlagEndEpoch)
 			if err != nil {
 				return err
 			}
 
-			lockIdStrs := strings.Split(lockIdsCombined, ",")
-			lockIds := []uint64{}
-			for _, lockIdStr := range lockIdStrs {
-				lockId, err := strconv.ParseUint(lockIdStr, 10, 64)
-				if err != nil {
-					return err
+			// if user doesn't provide at least one of the lock ids or owner, we don't have enough information to proceed.
+			if lockIdsCombined == "" && owner == "" {
+				return fmt.Errorf("either one of owner flag or lock IDs must be provided")
+
+				// if user provides lockIDs, use these lockIDs in our rewards estimation
+			} else if owner == "" {
+				for _, lockIdStr := range lockIdStrs {
+					lockId, err := strconv.ParseUint(lockIdStr, 10, 64)
+					if err != nil {
+						return err
+					}
+					lockIds = append(lockIds, lockId)
 				}
-				lockIds = append(lockIds, lockId)
+
+				// if no lockIDs are provided but an owner is provided, we query the rewards for all of the locks the owner has
+			} else if lockIdsCombined == "" {
+				lockIds = append(lockIds, ownerLocks...)
 			}
 
-			endEpoch, err := cmd.Flags().GetInt64(FlagEndEpoch)
+			// if lockIDs are provided and an owner is provided, only query the lockIDs that are provided
+			// if a lockID was provided and it doesn't belong to the owner, return an error
+			if len(lockIds) != 0 && owner != "" {
+				for _, lockId := range lockIds {
+					validInputLockId := contains(ownerLocks, lockId)
+					if !validInputLockId {
+						return fmt.Errorf("lock-id %v does not belong to %v", lockId, owner)
+					}
+				}
+			}
+
+			clientCtx, err := client.GetClientQueryContext(cmd)
 			if err != nil {
 				return err
 			}
+			queryClient := types.NewQueryClient(clientCtx)
+
+			if owner != "" {
+				queryClientLockup := lockuptypes.NewQueryClient(clientCtx)
 
-			res, err := queryClient.RewardsEst(cmd.Context(), &types.RewardsEstRequest{
+				res, err = queryClientLockup.AccountLockedLongerDuration(cmd.Context(), &lockuptypes.AccountLockedLongerDurationRequest{Owner: owner, Duration: time.Millisecond})
+				if err != nil {
+					return err
+				}
+				for _, lockId := range res.Locks {
+					ownerLocks = append(ownerLocks, lockId.ID)
+				}
+			}
+
+			// TODO: Figure out why some lock ids are good and some causes "Error: rpc error: code = Unknown desc = panic message redacted to hide potentially sensitive system info: panic"
+			// since owner checks have already been completed above, we switch the owner address to a random module account address since a blank owner panics.
+			// we should find a better way to circumvent this address validity check
+			if owner == "" {
+				owner = "osmo14kjcwdwcqsujkdt8n5qwpd8x8ty2rys5rjrdjj"
+			}
```

Comment:
We are more so worried about the case where owner = "" && len(lockIds != 0) right? Because the below RewardsEstRequest struct requires and owner even if its not used

Reply:
Fixed by adding logic to grpc_query.go